python/social_bot/envs/grocery_ground.py

```diff
@@ -63,22 +65,162 @@ def __init__(self, reward_shaping=False, random_goal=False, **kwargs):
             'coke_can', 'table', 'bookshelf', 'car_wheel',
             'plastic_cup', 'beer', 'hammer'
         ]
+        self._pos_list = list(itertools.product(range(-5, 5), range(-5, 5)))
+        self._pos_list.remove((0, 0))
         self.task_vocab = self.task_vocab + self._objects_in_world + self._objects_to_insert
 
-    def run(self, agent, world):
+    def setup(self, world, agent_name):
+        """
+        Setting things up during the initialization
+        """
+        self._world = world
+        self._agent = self._world.get_agent()
+        self._agent_name = agent_name
+        self._insert_objects(self._objects_to_insert)
+
+    def reset(self):
```

Comment:
To control the phase of reseting, which is necessary to avoid model collisions. And I thought it is readable to have such an explicitly initializing function.

Reply:
For every new episode, reset() and run() will be called exactly once. So there is no need to separate them. We should always try to make the interface simplest if possible.

Reply:
OK, seems we can remove task.reset(), by moving the teacher.teach() before world.step() in the env reset().